plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -692,14 +690,15 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     {
         DeltaLakeTableHandle table = (DeltaLakeTableHandle) tableHandle;
         DeltaLakeColumnHandle column = (DeltaLakeColumnHandle) columnHandle;
-        if (column.getProjectionInfo().isPresent()) {
-            return getColumnMetadata(column, null, true, null);
+        if (column.getProjectionInfo().isPresent() || column.getColumnType() == SYNTHESIZED) {
+            return getColumnMetadata(column, null, true, Optional.empty());
         }
+        DeltaTable deltaTable = DeltaTable.builder(table.getMetadataEntry(), table.getProtocolEntry()).build();
         return getColumnMetadata(
                 column,
-                getColumnComments(table.getMetadataEntry()).get(column.getBaseColumnName()),
-                getColumnsNullability(table.getMetadataEntry()).getOrDefault(column.getBaseColumnName(), true),
-                getGeneratedColumnExpressions(table.getMetadataEntry()).get(column.getBaseColumnName()));
+                deltaTable.getColumnComment(column.getBaseColumnName()),
```

Comment:
Here is also a good reason to expose the `DeltaLakeTableColumn` . We make several calls on the `deltaTable` variable always with the same parameter.

Reply:
I didn't want to expose another column class as we already have DeltaLakeColumnHandle class. I would focus on table class in this PR. Exposing the column class can be a follow-up.